core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -319,14 +331,54 @@ object Extraction {
       }
     }
 
+    def newTuple(root: JValue, mappings: List[Mapping]): Any = {
+      root match {
+        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+          val builtItems: Seq[Object] = items.zip(mappings).map({
+            case (i,m) =>
+              build(i, m).asInstanceOf[Object]
+          })
+          val tupleClass = Class.forName("scala.Tuple" + items.length)
+
+          val typedTupleConstructor = tupleClass.getConstructors()(0)
+          typedTupleConstructor.newInstance(builtItems: _*)
+
+        case JObject(items) if items.forall(_.name.startsWith("_")) =>
+          newTuple(JArray(items.map(_.value)), mappings)
+
+        case JArray(items) =>
+          throw new IllegalArgumentException("Cannot create a tuple of length " + items.length)
```

Comment:
It would be nice to move this condition above the `JObject` one so the two array possibilities are together.